libsolidity/interface/CompilerStack.cpp

```diff
@@ -1507,7 +1522,7 @@ void CompilerStack::generateIR(ContractDefinition const& _contract, bool _unopti
 	solAssert(m_stackState >= AnalysisSuccessful, "");
 
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
-	if (!compiledContract.yulIR.empty())
+	if (compiledContract.yulIR && !compiledContract.yulIR->empty())
```

Comment:
Checking the optional here is enough. Code that was actually generated will never be empty (you can assert that).

Same with `yulIROptimized` in `generateEVMFromIR()`.